nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/TestCommentsMigrator.java

```diff
@@ -279,7 +240,135 @@ public void testProbe() {
+        List<Comment> commentsForFile1 = propertyCommentManager.getComments(session, firstFileToComment.getId());
+        List<Comment> commentsForFile2 = propertyCommentManager.getComments(session, secondFileToComment.getId());
+
+        assertEquals(NB_COMMENTS, commentsForFile1.size() + commentsForFile2.size());
+        for (Comment comment : commentsForFile1) {
+            assertEquals(firstFileToComment.getId(), comment.getParentId());
+            assertNull(session.getDocument(new IdRef(comment.getId())).getACP().getACL(LOCAL_ACL));
+        }
+        for (Comment comment : commentsForFile2) {
+            assertEquals(secondFileToComment.getId(), comment.getParentId());
+            assertNull(session.getDocument(new IdRef(comment.getId())).getACP().getACL(LOCAL_ACL));
+        }
+
+        List<String> expectedLines = Arrays.asList( //
+                "Initializing: 0/-1", //
+                "Migrating comments from Relation to Property: 1/100", //
+                "Migrating comments from Relation to Property: 51/100", //
+                "Migrating comments from Relation to Property: 100/100", //
+                "Done Migrating from Relation to Property: 100/100");
+        assertEquals(expectedLines, progressLines);
+    }
+
+    /**
+     * @since 11.1
+     */
+    protected void createCommentsAsRelations() {
+        CommentManager relationCommentManager = new CommentManagerImpl(newConfig());
+        NuxeoPrincipal principal = session.getPrincipal();
+        // create some comments as relation
+        for (int i = 0; i < NB_COMMENTS; i++) {
+            DocumentModel comment = session.createDocumentModel(null, "comment_" + i, COMMENT_DOC_TYPE);
+            DocumentModel createdComment;
+            if (i % 2 == 0) {
+                createdComment = relationCommentManager.createComment(firstFileToComment, comment);
+            } else {
+                createdComment = relationCommentManager.createComment(secondFileToComment, comment);
+            }
+            notificationManager.addSubscription(principal.getName(), "notification" + i, createdComment, FALSE,
+                    principal, "notification" + i);
+        }
+
+        session.save();
+        transactionalFeature.nextTransaction();
+    }
+
+    /**
+     * @since 11.1
+     */
+    protected MigrationContext getMigrationContext(List<String> progressLines) {
+        return new MigrationContext() {
+            @Override
+            public void reportProgress(String message, long num, long total) {
+                String line = message + ": " + num + "/" + total;
+                progressLines.add(line);
+            }
+
+            @Override
+            public void requestShutdown() {
+            }
+
+            @Override
+            public boolean isShutdownRequested() {
+                return false;
+            }
+        };
+    }
+
+    /**
+     * @since 11.1
+     */
+    protected void mockUnsecuredPropertyComments() {
```

Comment:
This method makes me think that you need to have a working code after applying the HF without having to migrate to secured properties.